### PR TITLE
Add interfaces for IPlugin::localizedName() and IPlugin::master().

### PR DESCRIPTION
**Pull request by Holt59**
**TL;DR;** Adds bindings for `IPlugin::localizedName()` and `IPlugin::master()`.

This may looks like a lots of change to add two small bindings but I spent 4 hours trying to get this working... Basically, these two methods have default behavior, that must be specified at some point, but since the `IPlugin` custom codes are in macro, then it is not possible to only have them in the top-level `bpy::class_<IPluginWrapper, ...>` but we need them in each sub-class.

I also add to clear exception from Python since `localizedName()` recurses into `name()`, so `name()` would fail due to an error from `get_override()`, which is why I now have two `GILock`. I had to use a lambda here because `bpy::override` does not have a default constructor, and I did not want to modify `GILock` to allow for "unlock" or something like that.

### CODE REVIEW

**Comment by AnyOldName3** [comment]
I thought `get_override` was supposed to return a `nullptr` or something like that when there wasn't anything defined, in which case we could just test for that and call the base class implementation instead. Don't we already have something with a default implementation that basically does that?

**Comment by Holt59** [comment]
> I thought `get_override` was supposed to return a `nullptr` or something like that when there wasn't anything defined, in which case we could just test for that and call the base class implementation instead. Don't we already have something with a default implementation that basically does that?

That's what `basicWrapperFunctionImplementationWithDefault` does and what's used in the commit but that does not work if you only expose `localizedName` in the `IPlugin` bindings because then `get_override` does return something, which is basically a call to itself, so you end-up in an infinite recursion. The only way to prevent `get_override` from returning something when there is nothing is to expose `localizedName` in every plugin classes, not just in `IPlugin`.

**Review by Al12rs — APPROVED** [review]
